instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java

```diff
@@ -19,45 +17,50 @@
 import java.util.stream.Collectors;
 import java.util.stream.StreamSupport;
 
-public abstract class Instrumenter<REQUEST, RESPONSE> {
+public /* sealed */ class Instrumenter<REQUEST, RESPONSE> {
+
+  public static <REQUEST, RESPONSE> InstrumenterBuilder<REQUEST, RESPONSE> newBuilder(
+      OpenTelemetry openTelemetry, String instrumentationName) {
+    return new InstrumenterBuilder<>(openTelemetry, instrumentationName);
+  }
 
   private final Tracer tracer;
   private final SpanNameExtractor<? super REQUEST> spanNameExtractor;
+  private final SpanKind spanKind;
   private final StatusExtractor<? super REQUEST, ? super RESPONSE> statusExtractor;
   private final List<AttributesExtractor<? super REQUEST, ? super RESPONSE>> extractors;
+  private final ErrorCauseExtractor errorCauseExtractor;
 
-  protected Instrumenter(
+  Instrumenter(
       Tracer tracer,
       SpanNameExtractor<? super REQUEST> spanNameExtractor,
+      SpanKind spanKind,
```

Comment:
I do know that we _have to_ support customizing span kind / error cause but was hoping to not have too many features in one PR. But I guess moving spanKind was important for the idea of removing the inheritance. Can you Make it a `SpanKindExtractor<? super REQUEST>`?